xbmc/pvr/PVRActionListener.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ *      Copyright (C) 2005-2015 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "PVRActionListener.h"
+
+#include "Application.h"
+#include "ApplicationMessenger.h"
+#include "guilib/Key.h"
+#include "guilib/GUIWindow.h"
+#include "guilib/LocalizeStrings.h"
+#include "guilib/GUIWindowManager.h"
+#include "dialogs/GUIDialogNumeric.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "settings/AdvancedSettings.h"
+#include "settings/Settings.h"
+#include "utils/log.h"
+#include "utils/StringUtils.h"
+
+#include "pvr/PVRManager.h"
+#include "pvr/channels/PVRChannelGroupsContainer.h"
+
+using namespace PVR;
+
+CPVRActionListener::CPVRActionListener()
+{
+}
+
+CPVRActionListener &CPVRActionListener::Get()
+{
+  static CPVRActionListener instance;
+  return instance;
+}
+
+bool CPVRActionListener::OnAction(const CAction &action)
+{
+  switch (action.GetID())
+  {
+    case ACTION_PVR_PLAY:
+    case ACTION_PVR_PLAY_TV:
+    case ACTION_PVR_PLAY_RADIO:
+    {
+      // pvr not active yet, show error message
+      if (!g_PVRManager.IsStarted())
+      {
+        CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Warning, g_localizeStrings.Get(19045), g_localizeStrings.Get(19044));
```

Comment:
Since we register this listener when the manager is started I think we could remove this check.